### ISSUE TITLE
Make permissions API match PushNotificationIOS

### ISSUE THREAD

**Issue by ericvicenti** [mattotodd/react-native-addressbook-ios#1]
Hey, thanks for starting this module!

It would be great if the permissions API for address book matched the ones in React Native:

https://facebook.github.io/react-native/docs/pushnotificationios.html

**Comment by mattotodd** [comment]
Agreed

**Comment by rt2zz** [comment]
There are a couple of differences with the addressbook permissions,
1. The permission state is trinary vs push notifications which is boolean
2. There is only 1 permission for addressbook vs multiple

So maybe the API could be:
`checkPermission` & `requestPermission`

We could have the returned value be [undefined/false/true] rather than setting constants, but I think having undefined & false be distinct values may confuse some users. Thoughts?

I will make a PR for this later in combination with upgrades to RN 0.4.*

**Comment by mattotodd** [comment]
i think using the singular "permission"  makes sense.

Does Push's checkPermissions not distinguish the difference between undetermined and denied?

I'm fine with using undefined for the undetermined case as long as it is documented. (assuming there will never be a 4th status)

**Comment by rt2zz** [comment]
Ya push will either return true or false for all permissions. Perhaps they expect the application to store the meta about if/when they already asked for permission?

Another wierd thing is there is no callback on requestPermissions, so maybe this is not the best ideal for us to compare against.

**Comment by ericvicenti** [comment]
I still wish the API matched. Even if there is only one permission to access right now- will it always be that way?

If the React Native PushNotif API isn't ideal, maybe we could change it. It would be awesome if it were consistent, one way or the other.

**Comment by mattotodd** [comment]
@ericvicenti do you have another good example of requesting permission in react native ?   Otherwise there may not be a preferred way as of yet.  I have to agree with @rt2zz that no callback in requestPermissions is odd because it doesn't tell you what happened.

**Comment by ericvicenti** [comment]
A callback is a great idea. I don't think we get the results of the permission request for push notifs, which is probably why it is missing.

I just want to make sure we develop a preferred way together and do our best to keep the APIs in sync. I'm not sure of anything else which does explicit permissions requests.

**Comment by mattotodd** [comment]
Ok, good enough.  I will change the names of the calls to match push (pluralize).  I haven't looked at the actual push code yet, but I'll see if I have time this weekend to add the callback there as well.  At that point we should be in sync.  Sound good @ericvicenti ?

**Comment by ericvicenti** [comment]
Perfect, thank you! Now we have a solid pattern that should be usable by any new permission-checking modules.

cc @vjeux @tadeuzagallo

**Comment by mattotodd** [comment]
@ericvicenti @vjeux  do you guys have a preference for returning the Boolean/undefined verses constants? See thread above

**Comment by mattotodd** [comment]
@ericvicenti  I've made the api name changes,  and return a permissions object like Push does,  but I am still returning strings because there are currently three possible statuses (denied, authorized, undetermined).

https://github.com/mattotodd/react-native-addressbook-ios/commit/584b1686b400c0423f4b14db7ff731a883c5ea05

If you have never requested permission,  and call checkPermissions, it will come back as undetermined. Which is different than denied.  If a user had denied access,  calling requestPermissions subsequent times will not ask the user for permission again. Once denied,  the user must go into their notification settings and authorize access.  For this reason, I thought it made more sense to allow for more than a boolean response.

Closing for now, but if you'd like me to revisit just add a comment and reopen.
